### PR TITLE
nate/fix/LOOP-1927/manual-glucose-entry

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-1927

require the CGM manager to interpret glucose sample using its managed user thresholds

### CODE REVIEW

File: LoopKitUI/CGMManagerUI.swift [hunk 1]
```diff
@@ -22,6 +22,9 @@ public protocol CGMManagerUI: CGMManager, DeviceManagerUI {
     
     /// the completed percent of the progress bar to display in the status bar
     var cgmLifecycleProgress: DeviceLifecycleProgress? { get }
+    
+    /// gets the type of a glucose sample using the CGM manager managed glucose thresholds
+    func glucoseValueType(for glucose: GlucoseSampleValue) -> GlucoseValueType?
```
Comment (ps2) [inline LoopKitUI/CGMManagerUI.swift]: "ValueType" is quite vague. Looking at what GlucoseValueType contains, maybe `GlucoseRangeCategory`?

**Review by ps2 — APPROVED** [review]
LGTM. Just a naming suggestion.

**Comment by nhamming** [comment]
@ps2 Updated with the suggested naming.

Also, included a `Signal Loss` mode with alert as requested from design to assist with design review.

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**